### DISCUSSION TITLE
Pull Plots for systematics

### DISCUSSION THREAD

**Discussion by ssaha1234** [scikit-hep/pyhf#1507] [Q&A]
Dear all, 
     I am new to pyhf and I have recently made and run the json file using the pyhf workspace with my channels and samples. I have added my systematic uncertainties as histosys in the modifiers. 
I was wondering if anyone would know how I can start making the pull plot for various systematic uncertainties. Any basic idea about starting off would be great ! 

Best, 
Shreya

**Comment by matthewfeickert** [comment]
:wave: Hi @ssaha1234. `pyhf` doesn't try to focus on the visualizations, and we generally point people to @alexander-held's [`cabinetry`](https://github.com/alexander-held/cabinetry) for these sorts of things. That being said, we do have an example notebook of how you can make a pull plot with very little code: https://github.com/scikit-hep/pyhf/blob/106263debda987ecee403aeeab06726de4e8865e/docs/examples/notebooks/pullplot.ipynb

(I think that @kratsg and @eschanet _might_ have made some other scripts that they could share if they exist.)

**Reply by alexander-held** [reply]
`cabinetry` implements [`cabinetry.visualize.pulls`](https://cabinetry.readthedocs.io/en/latest/api.html#cabinetry.visualize.pulls) for this purpose, which can be used together with [`cabinetry.fit.fit`](https://cabinetry.readthedocs.io/en/latest/api.html#cabinetry.fit.fit) to obtain a `FitResults` object in the right format (can also be built by hand, it is just a named tuple). This [example notebook](https://nbviewer.jupyter.org/github/cabinetry/cabinetry-tutorials/blob/master/example.ipynb) includes these functions. The plot itself is done in [`cabinetry.contrib.matplotlib_visualize.pulls `](https://cabinetry.readthedocs.io/en/latest/api.html#cabinetry.contrib.matplotlib_visualize.pulls) via `matplotlib`.

**Reply by ssaha1234** [reply]
Dear @matthewfeickert and @alexander-held , 
    Thank you very much for all the details and pointing me to the examples, I will definitely try it out, and let you know how it goes ! 
Best, 
Shreya
